lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -36,6 +36,21 @@ pub struct GelfDeserializerConfig {
     pub gelf: GelfDeserializerOptions,
 }
 
+/// Configures the decoding validation mode.
+#[configurable_component]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+#[serde(rename_all = "snake_case")]
+pub enum ValidationMode {
+    /// Uses strict validation that closely follows the GELF spec.
+    #[default]
+    Strict,
+
+    /// Uses more relaxed validation similar to the [Go SDK][implementation].
+    ///
+    /// [implementation]: https://github.com/Graylog2/go-gelf/blob/v2/gelf/reader.go
```

Comment:
Not sure if we directly compare with Go SDK:

```suggestion
	/// Uses more relaxed validation that skips strict GELF specification checks.
    ///
    /// This mode will not treat specification violations as errors, allowing the decoder
    /// to accept messages from sources that don't strictly follow the GELF spec.
```

Feel free to decline this or recommend something else.

Reply:
Thank you, I've accepted the change